app/metrics/route.ts

```diff
@@ -0,0 +1,48 @@
+import { register, Gauge } from 'prom-client';
+import { calcMetrics } from '../utils';
+import { NextRequest } from 'next/server';
+
+export const dynamic = 'force-dynamic'; // defaults to auto
+
+let gauges: { [key: string]: Gauge } = {};
+
+export async function GET(request: NextRequest) {
+  const now = new Date();
+  const oneHourAgo = new Date();
+  oneHourAgo.setHours(now.getHours() - 1);
+
+  let metrics;
+  try {
+    metrics = await calcMetrics([oneHourAgo, now], request.nextUrl.origin);
+  } catch (error) {
+    console.error('Failed to calculate metrics:', error);
+    return new Response('Internal Server Error', { status: 500 });
+  }
+  for (let chainMetrics of metrics) {
+    for (let metric of chainMetrics) {
+      if (metric.stat == 0) {
+        continue
+      }
+
+      for (let stat of metric.data) {
+        let gauge = gauges[metric.key] ?? new Gauge({
+          name: metric.key,
+          help: metric.category,
+          labelNames: ['chainName', 'statName']
+        });
+        gauge.labels(metric.chain, stat.name).set(stat.value);
+        gauges[metric.key] = gauge
+      }
+    }
+  }
+
+  let defaultMetrics = await register.metrics();
+
+  return new Response(defaultMetrics, {
+    headers: {
+      'Cache-Control': 'no-store',
```

Comment:
do we not want it cached? seems like it might be worth caching for a bit?

Reply:
The caching of metrics seems wrong to me. It's something that always changing and the collector should always want to get the most up-to-date data points.

FYI we cache the responses from the rpc provider about the block information.